src/components/PeopleTable.tsx

```diff
@@ -1,682 +1,141 @@
-export const PeopleTable = () => {
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Person } from '../types';
+import { PersonItem } from './PersonItem/PersonItem';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  people: Person[];
+  slug: string | undefined;
+  isLoading: boolean;
+};
+
+export const PeopleTable: React.FC<Props> = ({ people, slug, isLoading }) => {
+  const [searchParams] = useSearchParams();
+
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+  const categoryFilter = ['Name', 'Sex', 'Born', 'Died'];
+
+  const preparePeople = () => {
+    let newPeople = [...people];
+
+    if (query) {
+      const normalizedQuery = query.trim().toLowerCase();
+
+      newPeople = newPeople.filter(person => {
+        return person.name.toLowerCase().includes(normalizedQuery)
+          || (person.motherName || '').toLowerCase().includes(normalizedQuery)
+          || (person.fatherName || '').toLowerCase().includes(normalizedQuery);
+      });
+    }
+
+    if (sex) {
+      newPeople = newPeople.filter(person => person.sex === sex);
+    }
+
+    if (centuries.length) {
+      newPeople = newPeople.filter(person => {
+        return centuries.includes(Math.ceil(person.born / 100).toString());
+      });
+    }
+
+    if (sort) {
+      newPeople = newPeople.sort((a, b) => {
+        switch (sort) {
+          case 'Name':
+            return a.name.localeCompare(b.name);
+          case 'Sex':
+            return a.sex.localeCompare(b.sex);
+          case 'Born':
+            return a.born - b.born;
+          case 'Died':
+            return a.died - b.died;
+          default:
+            return 0;
+        }
+      });
+    }
+
+    if (order === 'desc') {
+      newPeople = newPeople.reverse();
+    }
+
+    return newPeople;
+  };
```

Comment:
Memoize the result of this function instead of calling it in every render. (useMemo)

```suggestion
  const preparePeople = () => {
    let newPeople = [...people];

    if (query) {
      const normalizedQuery = query.trim().toLowerCase();

      newPeople = newPeople.filter(person => {
        return person.name.toLowerCase().includes(normalizedQuery)
          || (person.motherName || '').toLowerCase().includes(normalizedQuery)
          || (person.fatherName || '').toLowerCase().includes(normalizedQuery);
      });
    }

    if (sex) {
      newPeople = newPeople.filter(person => person.sex === sex);
    }

    if (centuries.length) {
      newPeople = newPeople.filter(person => {
        return centuries.includes(Math.ceil(person.born / 100).toString());
      });
    }

    if (sort) {
      newPeople = newPeople.sort((a, b) => {
        switch (sort) {
          case 'Name':
            return a.name.localeCompare(b.name);
          case 'Sex':
            return a.sex.localeCompare(b.sex);
          case 'Born':
            return a.born - b.born;
          case 'Died':
            return a.died - b.died;
          default:
            return 0;
        }
      });
    }

    if (order === 'desc') {
      newPeople = newPeople.reverse();
    }

    return newPeople;
  };
```